src/rulesets/Base/array.jl

```diff
@@ -18,13 +18,25 @@ end
 
 # Don't worry about projection here. The data passes straight through, so if a cotangent has
 # the wrong type for some reason, it must be the fault of another rule somewhere.
-function rrule(::typeof(Base.vect), X::Vararg{T,N}) where {T,N}
+function rrule(::typeof(Base.vect), X::Vararg{T, N}) where {T, N}
     function vect_pullback(ȳ)
+        X̄ = ntuple(n -> ȳ[n], N)
         return (NoTangent(), NTuple{N}(ȳ)...)
     end
     return Base.vect(X...), vect_pullback
 end
 
+# Numbers need to be projected because they don't pass straight through the function.
+# More generally, we would ideally project everything.
+function rrule(::typeof(Base.vect), X::Vararg{Number, N}) where {N}
```

Comment:
Should be safe to project arrays too, and mixtures (if anyone does that):
```suggestion
function rrule(::typeof(Base.vect), X::Vararg{Union{Number,AbstractArray{<:Number}}, N}) where {N}
```